pkg/repository/udmrepo/kopialib/repo_init.go

```diff
@@ -185,15 +286,151 @@ type byteBuffer struct {
 	buffer []byte
 }
 
-func (b *byteBuffer) Write(p []byte) (n int, err error) {
+type byteBufferReader struct {
+	buffer []byte
+	pos    int
+}
+
+func (b *byteBuffer) Write(p []byte) (int, error) {
 	b.buffer = append(b.buffer, p...)
 	return len(p), nil
 }
 
+func (b *byteBuffer) WriteTo(w io.Writer) (int64, error) {
+	n, err := w.Write(b.buffer)
+	return int64(n), err
+}
+
 func (b *byteBuffer) Reset() {
 	b.buffer = nil
 }
 
 func (b *byteBuffer) Length() int {
 	return len(b.buffer)
 }
+
+func (b *byteBuffer) Reader() io.ReadSeekCloser {
+	return &byteBufferReader{buffer: b.buffer}
+}
+
+func (b *byteBufferReader) Close() error {
+	return nil
+}
+
+func (b *byteBufferReader) Read(out []byte) (int, error) {
+	if b.pos == len(b.buffer) {
+		return 0, io.EOF
+	}
+
+	copied := copy(out, b.buffer[b.pos:])
+	b.pos += copied
+
+	return copied, nil
+}
+
+func (b *byteBufferReader) Seek(offset int64, whence int) (int64, error) {
+	newOffset := b.pos
+
+	switch whence {
+	case io.SeekStart:
+		newOffset = int(offset)
+	case io.SeekCurrent:
+		newOffset += int(offset)
+	case io.SeekEnd:
+		newOffset = len(b.buffer) + int(offset)
+	}
+
+	if newOffset < 0 || newOffset > len(b.buffer) {
+		return -1, errors.New("invalid seek")
+	}
+
+	b.pos = newOffset
+
+	return int64(newOffset), nil
+}
+
+var funcGetParam = maintenance.GetParams
+
+func writeInitParameters(ctx context.Context, repoOption udmrepo.RepoOptions, logger logrus.FieldLogger) error {
+	r, err := openKopiaRepo(ctx, repoOption.ConfigFilePath, repoOption.RepoPassword, nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		c := r.Close(ctx)
+		if c != nil {
+			logger.WithError(c).Error("Failed to close repo")
+		}
+	}()
+
+	params, err := funcGetParam(ctx, r)
+	if err != nil {
+		return errors.Wrap(err, "error getting existing maintenance params")
+	}
+
+	if params.Owner == backend.RepoOwnerFromRepoOptions(repoOption) {
```

Comment:
Should we consider checking if the actual maintenance parameters match the desired state, not just the owner field ? What if the owner matches but maintenance intervals have changed ?